ha_lineairdb.cc

```diff
@@ -751,107 +747,85 @@ int ha_lineairdb::is_primary_key_type_int() {
 /**
  * @brief Set the current key of the row
  *  which is requested by the query.
- *  The key is formatted as the following:
- *    "table-[TABLENAME]-key-[KEY_STRING]"
- *
+ * 
  * @param key
- *  By default, the part KEY_STRING is constructed from the fields of the
+ * By default, the key is constructed from the fields of the
  * specified row, except for DELETE queries. For DELETE queries, pass the
- * key-object as `key` to extract KEY_STRING desired to delete.
- *
- * @WANTFIX:
- * The delimiter "-" here we use, is incomplete.
- * This char maybe exist in argument `table_name` and `key_name`.
+ * argument to extract key desired to delete.
  */
 void ha_lineairdb::set_current_key(const uchar* key) {
-  current_key_.length(0);
+  current_key_.clear();
   {  // TABLE_NAME
-    current_key_.append("table-");
     const auto& table_name = table->s->table_name;
-    current_key_.append(table_name.str, table_name.length);
+    ldbField.set_lineairdb_field(table_name.str, table_name.length);
+    current_key_ = ldbField.get_lineairdb_field();
   }
   {
     // KEY_NAME
-    current_key_.append("-key-");
-
     if (key != nullptr) {  // DELETE
       if (int int_bytes = is_primary_key_type_int()) {
-        // for integer type keys
-        int primary_key = 0;
-        for (int i = 0; i < int_bytes; i++) {
-          primary_key = primary_key | key[i] << sizeof(char) * i;
-        }
-        current_key_.append(std::to_string(primary_key));
+        /**
+         * @WANTFIX: need to use appropriate type for numeric primary key
+         * Currently, primary_key can only handle signed numeric keys up to 8 bytes
+         * Appropriate types must be selected for unsigned numbers.
+        */
+        long primary_key = ldbField.convert_bytes_to_numeric(key, int_bytes);
+        const std::string&& intKey = std::to_string(primary_key);
```

Comment:
const rvalue reference is useless.
See https://www.codesynthesis.com/~boris/blog/2012/07/24/const-rvalue-references/

Reply:
I need a better understanding of lvalue and rvalue reference.
I will change it to `const std::string intKey = std::to_string(primary_key);` for now.

Reply:
[IMO]
use `std::string&&`